### DISCUSSION TITLE
Highs::passModel() takes the model by value

### DISCUSSION THREAD

**Discussion by cvanaret** [ERGO-Code/HiGHS#2078] [Q&A] [answered]
Hi,
Is there a good reason why `Highs::passModel()` takes the model (`HighsModel` or `HighsLp`) by value, instead of (const) reference?
This means the model (cost, bounds, constraint matrix, Hessian) is copied when this function is called, which is expensive when the model is large.

**Comment by jajhall** [comment, marked as answer]
I didn't write this, but I believe that one reason is so that we can use `std::move` thus

> h.passModel(std::move(lpmodel));

It also allows us to do things like this

> HighsStatus Highs::passModel(HighsLp lp) {
  HighsModel model;
  model.lp_ = std::move(lp);
  return passModel(std::move(model));
}

when only a `HighsLp` instance is passed, and this couldn't be done if a const reference were passed. Maybe we could overload with additional methods

> Highs::passModel(const &HighsModel model)

> Highs::passModel(const &HighsLp lp)


Note that HiGHS operates on its own internal model instance, rather than the model that is passed to it, since it needs to be able to modify the user's model.

**Comment by cvanaret** [comment]
Got it, you make a copy and steal it, so that you can modify it internally. Thanks for the quick reply!
